lib/TokenRequestCreator.js

```diff
@@ -15,8 +15,8 @@ class TokenRequestCreator {
         // While including a nonce is described as optional, it seems needed for the creation of unique token requests
         if (includeRandomNonce) {
             offset = buf.writeUint8(32, offset);
-            const nonce = crypto.randomBytes(16).toString('hex');
-            offset += buf.write(nonce, offset);
+            const nonce = crypto.randomBytes(32);
+            offset += nonce.copy(buf, offset, 0, 32);
```

Comment:
Alternatively,

```suggestion
            crypto.randomFillSync(buf, offset, 32);
            offset += 32;
```

Reply:
Hi @michaelficarra 

thank you for your proposal. I just tried this change and couldn't get any validations working on Safari (Sonoma) anymore. It has been a while but I think I ran into this exact issue when trying similar strategies in the beginning. Without the change I get the flow to properly work.

Did you get the changed version to work in Safari or another browser with PAT?